### PR TITLE
feat: [SRM-14453]: Fix evaluation filter on SLO Listing page

### PR DESCRIPTION
**Pull request by deepesh-ui**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/slos/CVSLOListingPage.utils.ts [hunk 1]
```diff
@@ -561,7 +574,15 @@ export const getSLODashboardWidgetsParams = (
   pageNumber?: number,
   monitoredServiceIdentifier?: string
 ): SLODashboardWidgetsParams => {
-  const { monitoredService, search, sliTypes, sloRiskFilter, targetTypes, userJourney } = filterState
+  const { monitoredService, search, sliTypes, sloRiskFilter, targetTypes, userJourney, evaluationType } = filterState
```
Comment (karansaraswat19) [inline src/modules/85-cv/pages/slos/CVSLOListingPage.utils.ts:577]: can we take out the common part from `getSLODashboardWidgetsParams` and `getServiceLevelObjectivesRiskCountParams`

File: src/modules/85-cv/pages/slos/components/SLODashbordFilters/__tests__/SLODashbordFilters.test.tsx [hunk 2]
```diff
@@ -8,12 +8,14 @@
...
 import { render, screen } from '@testing-library/react'
 import { TestWrapper } from '@common/utils/testUtils'
+import * as FeatureFlag from '@common/hooks/useFeatureFlag'
 import { initialState } from '@cv/pages/slos/CVSLOListingPage.utils'
 import SLODashbordFilters from '../SLODashbordFilters'
 import { filterItemData } from './SLODashbordFilters.mock'
 
 describe('SLODashbordFilters', () => {
   test('should have all the filters', () => {
+    jest.spyOn(FeatureFlag, 'useFeatureFlag').mockReturnValue(true)
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/SLODashbordFilters/__tests__/SLODashbordFilters.test.tsx:18]: pass feature flag in the `TestWrapper` rather mocking the hook

File: src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx [hunk 3]
```diff
@@ -384,6 +384,7 @@ describe('CVSLOsListingPage', () => {
 
   describe('Filters', () => {
     test('should check whether all the filters are present', () => {
+      jest.spyOn(FeatureFlag, 'useFeatureFlag').mockReturnValue(true)
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx:387]: same here

File: src/modules/85-cv/pages/slos/components/SLODashbordFilters/__tests__/SLODashbordFilters.test.tsx (hunk 2 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/SLODashbordFilters/__tests__/SLODashbordFilters.test.tsx:18]: only need to mock in one test , so added inside test

File: src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx (hunk 3 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx:387]: only need to mock in one test , so added inside test

File: src/modules/85-cv/pages/slos/CVSLOListingPage.utils.ts (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/CVSLOListingPage.utils.ts:577]: The context of this PR is to make integrate evaluation type filter
Surely , will pick up as separate PR

Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/CVSLOListingPage.utils.ts]: Done

File: src/modules/85-cv/pages/slos/components/SLODashbordFilters/__tests__/SLODashbordFilters.test.tsx (hunk 2 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/slos/components/SLODashbordFilters/__tests__/SLODashbordFilters.test.tsx:18]: No need to mock itself, you can simply pass the feature flag and boolean under `defaultFeatureFlagValues` prop. It is simple

```ts
defaultFeatureFlagValues?: Partial<Record<FeatureFlag, boolean>>
```

**Review by praneshg239 — DISMISSED** [review]

**Review by karansaraswat19 — APPROVED** [review]

**Review by praneshg239 — APPROVED** [review]

**Review by karansaraswat19 — APPROVED** [review]

**Review by praneshg239 — APPROVED** [review]

**Pull request merged**